src/mbgl/util/mapbox.cpp

```diff
@@ -39,6 +39,20 @@ std::string normalizeGlyphsURL(const std::string& url, const std::string& access
     return normalizeURL(url, accessToken);
 }
 
+std::string normalizeTileURL(const std::string& url) {
+    if (url.compare(url.length() - 4, 4, ".png") == 0)
```

Comment:
Fixed gl-js support for other image formats in https://github.com/mapbox/mapbox-gl-js/commit/241a07d0607447cbc393f2ccfab3f03e765ced13.